app/routers/organizations.py

```diff
@@ -1,15 +1,57 @@
 from fastapi import APIRouter, status, HTTPException
 from database import client
-from models import Organization
+from models import Organization, OrganizationResponse
+import secrets
+import string
+from fastapi.encoders import jsonable_encoder
 
-router = APIRouter()
+router = APIRouter(prefix="/organizations", tags=["Organizations"])
 
 
-@router.get("/check-auth-token/{api_key}", response_model=Organization)
+@router.post("/", response_model=OrganizationResponse)
+async def create_organization(org: Organization):
+    org = jsonable_encoder(org)
+    if org["name"] is not None:
+
+        # create an API key
+        key = "".join(
```

Comment:
I think if you move this first key generation inside the loop before the `if`, you won't need the `else` at all